lte/gateway/c/oai/tasks/sgw/pgw_handlers.c

```diff
@@ -606,6 +632,77 @@ uint32_t pgw_handle_nw_initiated_bearer_deactv_req(
     pthread_mutex_unlock(&hashtblP->lock_nodes[i]);
     i++;
   }
+
+  if (send_reject) {
+    print_bearer_ids_helper(invalid_ebi, no_of_bearers_rej);
+    OAILOG_INFO(
+      LOG_PGW_APP,
+        "Sending dedicated bearer activation reject to NW\n");
+    //TODO-Uncomment once implemented at PCRF
+    /*rc = send_dedicated_bearer_deactv_rsp(invalid_ebi,
+        REQUEST_REJECTED);*/
+    OAILOG_FUNC_RETURN(LOG_PGW_APP, RETURNerror);
+  }
+
+  if (!is_imsi_found) {
+    OAILOG_ERROR(
+      LOG_PGW_APP,
+      "Wrong IMSI (%s) received in pgw_nw_init_deactv_bearer_request \n",
+      bearer_req_p->imsi);
+    OAILOG_INFO(
+      LOG_PGW_APP,
+      "Sending dedicated_bearer_deactv_rsp with REQUEST_REJECTED "
+      "cause to NW\n");
+    //Send Reject to PCRF
+    //TODO-Uncomment once implemented at PCRF
+    /*rc = send_dedicated_bearer_deactv_rsp(bearer_req_p->ebi,
+        REQUEST_REJECTED);*/
+    OAILOG_FUNC_RETURN(LOG_PGW_APP, RETURNerror);
+  } else if ((bearer_req_p->lbi != 0) && (!is_lbi_found)){
+    OAILOG_ERROR(
+      LOG_PGW_APP,
+      "Wrong LBI (%d) received in pgw_nw_init_deactv_bearer_request \n",
+      bearer_req_p->lbi);
+    OAILOG_INFO(
+      LOG_PGW_APP,
+      "Sending dedicated_bearer_deactv_rsp with REQUEST_REJECTED "
+      "cause to NW\n");
+    //Send Reject to PCRF
+    //TODO-Uncomment once implemented at PCRF
+    /*rc = send_dedicated_bearer_deactv_rsp(bearer_req_p->lbi,
+        REQUEST_REJECTED);*/
+    OAILOG_FUNC_RETURN(LOG_PGW_APP, RETURNerror);
+  }
```

Comment:
This needs some cleaning along with the block if(send_reject) {} above
The three places will eventually send the same response back to PCEF. Please restructure.

Reply:
done